### PR TITLE
Add helper class to simplify making branch-aware web calls

### PR DESCRIPTION
**Pull request by duncanp-sonar**
Prep for #3833

### CODE REVIEW

File: src/ConnectedMode/ServerQueryInfoProvider.cs [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    /// <remarks>Facade to simplify classes that need to make branch-aware calls.
+    /// The projectKey and server branch are provided by different interfaces. This interface
+    /// simplifies getting the information.</remarks>
+    internal interface IServerQueryInfoProvider
+    {
+        /// <summary>
+        /// Returns the projectKey and branch name for the current bound solution, or
+        /// (null, null) if the solution is not bound
+        /// </summary>
+        Task<(string projectKey, string branchName)> GetProjectKeyAndBranchAsync(CancellationToken token);
```
Comment (duncanp-sonar) [inline src/ConnectedMode/ServerQueryInfoProvider.cs:41]: The web calls to fetch the suppressions need to know the current project key and branch name, and need to handle the details not being available.

Originally, I'd put this functionality in the `SuppressionIssueStoreUpdater` class but it made it more complicated than necessary, so I broke it out into a separate class. It's likely that we could re-use it to simpligy other classes that call the server.

File: src/ConnectedMode.UnitTests/ServerQueryInfoProviderTests.cs [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+        public void MefCtor_CheckIsExported()
+        {
+            MefTestHelpers.CheckTypeCanBeImported<ServerQueryInfoProvider, IServerQueryInfoProvider>(
+                MefTestHelpers.CreateExport<IConfigurationProvider>(),
+                MefTestHelpers.CreateExport<IStatefulServerBranchProvider>());
+        }
+
+        [TestMethod]
+        [DataRow(SonarLintMode.Connected)]
+        [DataRow(SonarLintMode.Connected)]
```
Comment (bigfluffycookie) [inline src/ConnectedMode.UnitTests/ServerQueryInfoProviderTests.cs:43]: Is this a mistake or is there some reason to run the same thing twice

**Review by bigfluffycookie — APPROVED** [review]
Looks good, just one question regarding a test

File: src/ConnectedMode.UnitTests/ServerQueryInfoProviderTests.cs (hunk 2 above)
Comment (duncanp-sonar) [reply src/ConnectedMode.UnitTests/ServerQueryInfoProviderTests.cs:43]: Oops - mistake. Thanks!

**Pull request merged**